cmd/kubeadm/app/cmd/upgrade/apply_test.go

```diff
@@ -19,58 +19,12 @@ package upgrade
 import (
 	"io/ioutil"
 	"os"
-	"reflect"
 	"testing"
 
 	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
 )
 
-func TestSetImplicitFlags(t *testing.T) {
-	var tests = []struct {
-		name          string
-		flags         *applyFlags
-		expectedFlags applyFlags
-		errExpected   bool
-	}{
-		{
-			name: "if the new version is empty; it should error out",
-			flags: &applyFlags{
-				newK8sVersionStr: "",
-			},
-			expectedFlags: applyFlags{
-				newK8sVersionStr: "",
-			},
-			errExpected: true,
-		},
-	}
-	for _, rt := range tests {
-		t.Run(rt.name, func(t *testing.T) {
-			actualErr := SetImplicitFlags(rt.flags)
-
-			// If an error was returned; make newK8sVersion nil so it's easy to match using reflect.DeepEqual later (instead of a random pointer)
-			if actualErr != nil {
-				rt.flags.newK8sVersion = nil
-			}
-
-			if !reflect.DeepEqual(*rt.flags, rt.expectedFlags) {
-				t.Errorf(
-					"failed SetImplicitFlags:\n\texpected flags: %v\n\t  actual: %v",
-					rt.expectedFlags,
-					*rt.flags,
-				)
-			}
-			if (actualErr != nil) != rt.errExpected {
-				t.Errorf(
-					"failed SetImplicitFlags:\n\texpected error: %t\n\t  actual: %t",
-					rt.errExpected,
-					(actualErr != nil),
-				)
-			}
-		})
-	}
-}
-
```

Comment:
Should we move this test to NewCmdApply tests instead of removing?

Reply:
I'm not sure how `NewCmdApply()` can be testable since it calls `os.Exit(1)` in many places. 

One of the goals of https://github.com/kubernetes/kubernetes/pull/73135 was to minimize the code `NewCmdApply()` so that more code could be covered with tests without the need to introduce various "exiters". Also the flag itself is going to be removed. I'd rather test this code path in the test for [getK8sVersionFromUserInput()](https://github.com/kubernetes/kubernetes/pull/73135/files#diff-55e62cefda90d9f31865b8dd86ebf43fR49).

But anyway... the concern is valid. I'll rearrange the PR split to avoid code coverage decrease in the process.